pkg/apis/core/types_shoot.go

```diff
@@ -600,6 +609,18 @@ type KubeletConfigEvictionSoftGracePeriod struct {
 	NodeFSInodesFree *metav1.Duration
 }
 
+// KubeletConfigReserved contains reserved resources for daemons
+type KubeletConfigReserved struct {
+	// CPU is the reserved cpu.
+	CPU *resource.Quantity
+	// Memory is the reserved memory.
+	Memory *resource.Quantity
+	// EphemeralStorage is the reserved ephemeral-storage.
+	EphemeralStorage *resource.Quantity
+	// PID is the reserved process-ids.
+	PID *int64
```

Comment:
The kubelet seems to handle `pid` as a `resource.Quantity` just like the other resources (ref: https://github.com/kubernetes/kubernetes/blob/43fbe17dc682b703b0b663935a797244378f0aaf/cmd/kubelet/app/server.go#L1250-L1252)
I think, it would make sense to do the same in our API.
```suggestion
	PID *resource.Quantity
```